sqrt_arm64.s

```diff
@@ -0,0 +1,12 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include "textflag.h"
+
+// func Sqrt(x float64) float64
+TEXT ·Sqrt(SB),NOSPLIT,$0
+	FMOVD	x+0(FP), F0
+	FSQRTD	F0, F0
```

Comment:
@chewxy I don't think this is right. The package is a math32 package, so presumably the functions are for float32. This is a double precision instruction. The code above is similar.

Reply:
Aha! I'm not too well versed in ARM instructions (I know nothing... like Jon Snow). What should the correct instruction be?

@blackrez  care to chime in?

Reply:
I sent [this](https://github.com/gonum/gonum/pull/1256) to our internal math32. It is copied [from the stdlib math package](https://github.com/golang/go/blob/5bd145413a84be1afa74a82767384d9e224f7069/src/math/sqrt_arm64.s#L8-L12) with the change from double to single precision. I'd wait for @Kunde21 to chime in on that PR since while I think the FMOVS is correct, it may be better to use the FMOVD for performance reasons.

Reply:
Using `FSQRTD` is going to calculate using the double precision register (float64).  @kortschak is correct, the instruction you want here is `FSQRTS` for single precision (float32) or convert `FCVT` to and from double for the calculation.  

The `FMOV` instruction is not affected by the operand width like its x64 counterparts.  Best to use the same width for readability in arm64.

Reply:
@Kunde21. The reference is not completely clear on the behaviour of FMOV for this addressing mode (when taking into account the plan9 assembly syntax). Is `FMOVS   x+0(FP), F0` the same as `FMOV  Xd, Vn.D[1]    ; Top half of 128-bit to 64-bit` (wanting clarification for https://github.com/gonum/gonum/pull/1256.

Reply:
`FMOVS  x+0(FP), F0` is the same as the `FMOV Sd, Sn` variant and mirrors the `MOVSD` instruction from x86.  
`FMOV Xd, Vn.D[1] ; Top half of 128-bit to 64-bit` is more the sibling of `MOVHLPD Xn, Xd` from x86, though not exactly.  The move/load/store instructions of arm64 are much more flexible than the x86 instruction set, especially for vector registers.

Reply:
Thanks